### PR TITLE
chore(hardware): add opentrons hardware package to ot 2 build

### PR DESCRIPTION
**Pull request by fsinapi**
# Overview

This PR, in conjunction with https://github.com/Opentrons/buildroot/pull/209 adds the opentrons_hardware package to OT-2 builds. This is the beginning of plans to port most of the hardware-dependent code in the `opentrons` package over to `opentrons_hardware`. As of now, the package cannot actually be imported on the OT-2.

# Test Plan

Uploaded the build that was created from both this and the corresponding buildroot PR to an OT-2. Everything started up as normal, the app could still talk to the robot, and HTTP endpoints responded as normal. Hitting a Flex-specific endpoint still gets filtered out, with the error detail "No module named 'can'". SSH'd in and checked that the `opentrons_hardware` package was on the filesystem as expected.

# Changelog

- Mark python-can as an optional dependency of the hardware package in setup.py
- Add buildroot configuration to install opentrons_hardware and generate a version like other packages
- All of the `setup-ot2` targets remove `python-can` instead of the entire `opentrons_hardware` package

# Review requests

# Risk assessment

### CODE REVIEW

File: hardware/Config.in [hunk 1]
```diff
@@ -0,0 +1,17 @@
+config BR2_PACKAGE_PYTHON_OPENTRONS_HARDWARE
+  bool "python-opentrons-system-server"
```
Comment (vegano1) [inline hardware/Config.in]: hardware server

File: hardware/Config.in [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  select BR2_PACKAGE_PYTHON_JSONSCHEMA # runtime
+  select BR2_PACKAGE_PYTHON_NUMPY # runtime
+  select BR2_PACKAGE_PYTHON_PYDANTIC # runtime
+  select BR2_PACKAGE_PYTHON_SERIAL # runtime
+  select BR2_PACKAGE_PYTHON_SYSTEMD # runtime
+  select BR2_PACKAGE_PYTHON_TYPING_EXTENSIONS # runtime
+  select BR2_PACKAGE_PYTHON_CLICK # runtime
+
+  help
+    Opentrons system HTTP server. Provides access to an OT-2 robot.
```
Comment (vegano1) [inline hardware/Config.in]: - hardware server
- OT-2 and Flex or maybe Opentrons robot?

**Review by vegano1 — APPROVED** [review]
This is awesome,
left 2 simple copypasta errors
but otherwise this looks good

File: hardware/Config.in (hunk 1 above)
Comment (fsinapi) [reply hardware/Config.in]: indeed

File: hardware/Config.in (hunk 2 above)
Comment (fsinapi) [reply hardware/Config.in]: 🤖

File: api/setup.py [hunk 3]
```diff
@@ -60,6 +60,7 @@ def get_version():
 PACKAGES = find_packages(where="src")
```
Comment (sfoster1) [inline api/setup.py]: can we see if we can make the extras thing transitive? like, add an extra to `api` that is `'flex-hardware'` that requires `opentrons_hardware["CAN"]`, an extra that is `ot2-hardware` that requires `opentrons_hardware` (maybe eventually `opentrons_hardware["SERIAL"]` or whatever). Then the pipfile for api installs `opentrons["flex-hardware"]`.

The reason for this is that ideally we don't require hardware support for default in the api server.

**Review by sfoster1 — CHANGES_REQUESTED** [review]
I'd love to see if we can bubble up the extras to something like "install hardware support for flex" on the `opentrons` package.

Also we should check that the OE build still works since that consumes the pipfile I believe.

File: api/setup.py (hunk 3 above)
Comment (fsinapi) [reply api/setup.py]: Yep this worked pretty nicely. Tested running all of the following and they seem to filter the packages correctly:
- `pipenv run pip install .` --> no hardware package
- `pipenv run pip install ".[ot2-hardware]"` --> opentrons_hardware without python-can
- `pipenv run pip install ".[flex-hardware]"` --> opentrons_hardware with python_can

**Review by sfoster1 — APPROVED** [review]
This is awesome!

**Comment by fsinapi** [comment]
Tested installing the Openembedded build on Flexy, everything is still in the right place and the server seems to be chugging along fine

**Pull request merged**